src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/ObjectFormatting/Formatters/DebuggerDisplay/DebuggerDisplayFormatter.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using static Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay.DebuggerDisplayParser;
+
+namespace Microsoft.Diagnostics.Monitoring.HostingStartup.ParameterCapturing.ObjectFormatting.Formatters.DebuggerDisplay
+{
+    internal static class DebuggerDisplayFormatter
+    {
+        internal record struct DebuggerDisplayAttributeValue(string Text, IList<Type> EncompassingTypes);
+
+        public static FormatterFactoryResult? GetDebuggerDisplayFormatter(Type objType, ObjectFormatterCache? formatterCache = null)
```

Comment:
Once we integrate this into callable code, please make sure to guard against exceptions from parsing.